runtime/src/impls.rs

```diff
@@ -282,6 +286,89 @@ impl IsGasMeteredCall for GasMeteredCallResolver {
 	}
 }
 
+/// Provides a cennznet version of doughnut dispatch verification
+pub struct CENNZnetDispatchVerifier;
+
+/// Helpers which are used by the DelegatedDispatchVerifier
+impl CENNZnetDispatchVerifier {
+	/// Checks if the doughnut holds a `cennznut` and if so, it returns the decoded `cennznut`
+	fn get_cennznut(doughnut: &<Runtime as frame_system::Trait>::Doughnut) -> Result<CENNZnut, &'static str> {
```

Comment:
These could be stand alone helper methods, not really required on `CENNZnetDispatchVerifier`, ideally would be implemented on the doughnut but seems not worth the extra effort